ASFAchievementManager/AchievementHandler.cs

```diff
@@ -21,53 +19,61 @@ public sealed class AchievementHandler : ClientMsgHandler {
 
 			switch (packetMsg.MsgType) {
 				case EMsg.ClientGetUserStatsResponse:
-					HandleGetUserStatsResponse(packetMsg);
+					ClientMsgProtobuf<CMsgClientGetUserStatsResponse> getAchievementsResponse = new(packetMsg);
+					Client.PostCallback(new GetAchievementsCallback(packetMsg.TargetJobID, getAchievementsResponse.Body));
 
 					break;
 				case EMsg.ClientStoreUserStatsResponse:
-					HandleStoreUserStatsResponse(packetMsg);
+					ClientMsgProtobuf<CMsgClientStoreUserStatsResponse> setAchievementsResponse = new(packetMsg);
+					Client.PostCallback(new SetAchievementsCallback(packetMsg.TargetJobID, setAchievementsResponse.Body));
 
 					break;
 			}
 
 		}
 
-		private void HandleGetUserStatsResponse(IPacketMsg packetMsg) {
-			if (packetMsg == null) {
-				ASF.ArchiLogger.LogNullError(nameof(packetMsg));
-				return;
-			}
-			ClientMsgProtobuf<CMsgClientGetUserStatsResponse> response = new ClientMsgProtobuf<CMsgClientGetUserStatsResponse>(packetMsg);
-			if (!Responses.TryAdd(response.Body.game_id, new StoredResponse {
-				Success = response.Body.eresult == 1,
-				Response = response.Body
-			})) {
-				ASF.ArchiLogger.LogGenericError(string.Format(Strings.ErrorFailingRequest, "GetAchievements"));
-			}
+		internal sealed class GetAchievementsCallback : CallbackMsg {
```

Comment:
You can make use of derivation with `GetAchievementsCallback` and `SetAchievementsCallback`. There is only difference of 3 lines (and one of them is a one-letter difference in a logged string)

Reply:
Good idea - I've made some changes, is it better now?